### PR TITLE
Game dynamic difficulty

### PR DESCRIPTION
**Pull request by maxschemann**
Die Schwierigkeit einer Karteikarte passt sich durch richtiges oder falsches Übersetzen nach unten oder oben an und wird in der DB geupdated

### CODE REVIEW

File: frontend/src/component/Game.tsx [hunk 1]
```diff
@@ -1,42 +1,92 @@
...
-            const rdmIndex = Math.floor(Math.random() * indexCards.length)
-            if (indexArray.indexOf(rdmIndex) === -1) {
-                indexArray.push(rdmIndex)
-                counter++
-            }
+    const submitTranslation = () => {
+        if (translation === nextCard.term2 && nextCard.difficulty.toString() === Difficulty[Difficulty.EASY]) {
+            pickNextCard()
+        }
+        if (translation !== nextCard.term2 && nextCard.difficulty.toString() === Difficulty[Difficulty.HARD]) {
```
Comment (Boltzmann) [inline frontend/src/component/Game.tsx:29]: Maybe I did not see that in a latter review or it bothers me only now: The code is very understandable. However, regarding terms: In my head card.term1 is the given language, card.term2 is the translation to be guessed. What about finding some term for term1/term2 that describes its intention? Like term1 => givenTerm and term2 => termToGuess. Maybe there is some other word for it, that makes reading and understanding easier.

File: frontend/src/component/GameOptions.tsx [hunk 2]
```diff
@@ -1,25 +1,45 @@
...
+                indexArray.push(rdmIndex)
+                counter++
+            }
+        }
+        return indexArray
+    }
+
+    const reorderCards = () => {
+        return randomIndexArray().map(randomIndex => indexCards[randomIndex])
+    }
```
Comment (Boltzmann) [inline frontend/src/component/GameOptions.tsx:32]: It looks to me, that randomIndexArray should give an Array with randomly distributed numbers (0-indexCards.length-1) that has a indexCard.length size. However, an element is pushed only if indexOf(rdmIndex) === -1? Also, the rdmIndex that is defined in randomIndexArray() is used in the map of randomIndexArray()? Could you please explain to me. If it is good, then I am curious to understand.

**Review by Boltzmann — COMMENTED** [review]
I have read all the changes and made around two comments. The second one - however - is so interesting to me, that I'd like to investigate: How do you shuffle the cards? I will think about this, but until I have discovered the mechanism, we should talk about this.

File: frontend/src/component/Game.tsx (hunk 1 above)
Comment (maxschemann) [reply frontend/src/component/Game.tsx:29]: givenTerm/ TermToGuess might be a good change, i'll think about that. Something like 'word'/ 'translation' would be easier to read for sure and it was like that before, but in the end I wanted to be able to just switch the terms and don't get any conflicts with meanings that way.

File: frontend/src/component/GameOptions.tsx (hunk 2 above)
Comment (maxschemann) [reply frontend/src/component/GameOptions.tsx:32]: -1 is a predefined return value of .indexOf, when the value isn't in the array yet, so it's just a check, if the index already exists.
After creating that index array, the cards are arranged in the same order. I'm open to feedback, that was just the first solution from the top of my head on how to make sure that there are no doublicates in the list.

**Review by Gadenko — APPROVED** [review]

**Pull request merged**